DeutschJozsaAlgorithm/Tasks.qs

```diff
@@ -162,14 +162,58 @@ namespace Quantum.Kata.DeutschJozsaAlgorithm {
         // Hint: represent f(x) in terms of AND and ⊕ operations
 
         // ...
+    } 
+    
+    //////////////////////////////////////////////////////////////////
+    // Part II. Deutsch-Jozsa Algorithm
+    //////////////////////////////////////////////////////////////////
+    
+    // Task 2.1. Deutsch-Jozsa algorithm implementation
+    // Inputs:
+    //      1) the number of qubits in the input register N for the function f
+    //      2) a quantum operation which implements the oracle |x⟩|y⟩ -> |x⟩|y ⊕ f(x)⟩, where
+    //         x is an N-qubit input register, y is a 1-qubit answer register, and f is a Boolean function
+    // You are guaranteed that the function f implemented by the oracle is either
+    // constant (returns 0 on all inputs or 1 on all inputs) or
+    // balanced (returns 0 on exactly one half of the input domain and 1 on the other half).
+    // Output:
+    //      true if the function f is constant
+    //      false if the function f is balanced
+    //
+    // Note: a trivial approach is to call the oracle multiple times:
+    //       if the values for more than half of the possible inputs are the same, the function is constant.
+    // Quantum computing allows to perform this task in just one call to the oracle; try to implement this algorithm.
+    operation DJ_Algorithm (N : Int, Uf : ((Qubit[], Qubit) => Unit)) : Bool {
+        
+        // Declare Bool variable in which the result will be accumulated;
+        // this variable has to be mutable to allow updating it.
+        mutable isConstantFunction = true;
+        
+        // ...
+
+        return isConstantFunction;
     }
     
     
+    // Task 2.2. Testing Deutsch-Jozsa algorithm
+    // Goal: use your implementation of Deutsch-Jozsa algorithm from task 3.1 to test
+    // each of the oracles you've implemented in part I for being constant or balanced.
+    operation DJ_Test () : Unit {
+        // Hint: you will need to use partial application to test oracles such as Oracle_Kth_Qubit and Oracle_ProductFunction;
+        // see task 2.3 for a description of how to do that.
+
+        // Hint: use the Fact function to assert that the return value of DJ_Algorithm operation matches the expected value
+
+        // DJ_Test appears in the list of unit tests for the solution; run it to verify your code.
+
+        // ...
+    }
+    
     //////////////////////////////////////////////////////////////////
-    // Part II. Bernstein-Vazirani Algorithm
+    // Part III. Bernstein-Vazirani Algorithm
     //////////////////////////////////////////////////////////////////
     
-    // Task 2.1. State preparation for Bernstein-Vazirani algorithm
+    // Task 3.1. State preparation for Bernstein-Vazirani algorithm
```

Comment:
I think it's better to keep this task in part II - state prep for both algorithms is the same, so we need to teach it before the first time learner needs it. So this should stay 2.1, and DJ algorithm and running it should be 2.2 and 2.3, respectively.